src/client/client.ts

```diff
@@ -0,0 +1,103 @@
+import { Event, ResponseEvent } from "core/events";
+import fetch, { Response } from "cross-fetch";
+import { EventResponse } from "client/response";
+import { HttpError, TimeoutError } from "client/errors";
+import { getErrorType } from "core/errors";
+import { intoEvent } from "core/utils";
+
+const httpResponseHandler = (event: Event) => (
+    response: Response
+): Response => {
+    if (response.status === 200) {
+        return response;
+    }
+    throw new HttpError(
+        response.statusText,
+        response.status,
+        response,
+        event.flowId,
+        event.id
+    );
+};
+
+const convertToEvent = (response: Response): Event => {
+    const eventJson = response.json();
+    return intoEvent(eventJson);
+};
+
+const convertToEventResponse = (event: Event): EventResponse => {
+    const eventNameAppend = event.name.slice(event.name.lastIndexOf(":") + 1);
+    return eventNameAppend === "response"
+        ? { event: new ResponseEvent(event) }
+        : {
+              event: new ResponseEvent(event),
+              errorType: getErrorType(eventNameAppend)
+          };
+};
+
+const timeoutFunc = async (
+    event: Event,
+    timeout: number,
+    promise: Promise<Response>
+): Promise<Response> => {
+    return new Promise((resolve, reject) => {
+        setTimeout(() => {
+            reject(new TimeoutError(event));
+        }, timeout);
+        promise.then(resolve, reject);
+    });
+};
+
+interface ClientConfig {
+    defaultTimeout: number;
+    fetchHandler: typeof fetch;
+}
+
+interface ParamClientConfig {
+    defaultTimeout?: number;
+    fetchHandler?: typeof fetch;
+}
+
+const BASE_CONFIG: ClientConfig = {
+    defaultTimeout: 30000,
+    fetchHandler: fetch
+};
+
+export class EventsClient {
+    private url: string;
+    private config: ClientConfig;
+
+    constructor(url: string, config: ParamClientConfig = BASE_CONFIG) {
+        this.url = url;
+        this.config = { ...BASE_CONFIG, ...config };
```

Comment:
Mesmo se eu sobrescrever a base config ela vai ser adicionada de qualquer jeito?

Se for assim, eu colocaria para o parâmetro config ser `config = { }`, pra não passar a impressão de que o `BASE_CONFIG` só virá como default